From e846b5b94549fa03b1bcca2bc1948b3a547e860d Mon Sep 17 00:00:00 2001
From: Andi Kleen <ak@linux.intel.com>
Date: Sun, 25 Apr 2021 16:36:42 -0700
Subject: [PATCH 39/39] kbuild/bloat-o-meter: Handle gcc LTO

gcc LTO can add .lto_priv postfixes to symbols. Ignore those
in bloat-o-meter to allow comparision of non LTO with LTO
kernels.

Signed-off-by: Andi Kleen <ak@linux.intel.com>
---
 scripts/bloat-o-meter | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/bloat-o-meter b/scripts/bloat-o-meter
index dcd8d8750b8b..ec5ece4eed65 100755
--- a/scripts/bloat-o-meter
+++ b/scripts/bloat-o-meter
@@ -37,7 +37,7 @@ def getsizes(file, format):
                 if name.startswith("__addressable_"): continue
                 if name == "linux_banner": continue
                 # statics and some other optimizations adds random .NUMBER
-                name = re_NUMBER.sub('', name)
+                name = re_NUMBER.sub('', name).replace(".lto_priv", "")
                 sym[name] = sym.get(name, 0) + int(size, 16)
     return sym
 
-- 
2.30.2

